tests/CMakeLists.txt

```diff
@@ -1,15 +1,31 @@
-set(libs
-    mbedtests
-    mbedtls
-)
-
 # Set the project root directory if it's not already defined, as may happen if
 # the tests folder is included directly by a parent project, without including
 # the top level CMakeLists.txt.
 if(NOT DEFINED MBEDTLS_DIR)
     set(MBEDTLS_DIR ${CMAKE_SOURCE_DIR})
 endif()
 
+# For testing purposes, the Mbed TLS libraries are expected to rely on common
+# test code, for example, of the testing of the crypto library backed-up by
+# PSA drivers.
+# This dependency cannot be defined as part of the library build system as we
+# want libraries to be completely independant of test code and test build
+# system. For the unit tests, the dependencies are thus defined here.
+# We specify (through the target_link_libraries command) each test executable to
+# link against mbedtls, mbedx509, mbedcrypto and mbedtests in that order. The
+# order is important as:
```

Comment:
Well, I am not completely sure to understand your concern. I've tried to explain the need for ordering but failed to make it clear enough it seems.
The root problem here is that it seems there is no "linker independent" proper solution, thus basically no proper solution, to the problem of mutual/cyclic dependencies between libraries. It sounds a bit incredible but that the way it is .
The usual nasty way of dealing with mutual/cyclic dependencies is to repeat several time in an "appropriate" order the libraries in the link command line like: -llibA -llibB -llibA -llibB ... and you end up with the above kind of obscure comments to explain this nightmare.
Eventually, I think I will move to a list of objects for the test common code. This will avoid the library cyclic dependencies issue and should be good enough at least for the time being. Maybe, that we will feel the need for proper libraries for test drivers but we will see at that point.

Reply:
I know that there's no portable way to deal with cyclic dependencies. But the commit message and the comment together seem contradictory: on the one hand there are cyclic dependencies, but on the other hand the link needs to happen in this particular order because the dependencies are in this order.

Reply:
And indeed this is a problem that I hadn't thought of, and I agree to do whatever is simple and works now, and worry later if it isn't enough. I just want it to be clear what we're doing now and what assumptions we're making.